### ISSUE TITLE
newrelic_synthetics_script_monitor fails with `This field is deprecated! Please use relatedEntities instead`

### ISSUE THREAD

**Issue by sirianni** [newrelic/terraform-provider-newrelic#2485]
### Affected Resource(s)
- `newrelic_synthetics_script_monitor`

### Terraform Configuration
```terraform
resource "newrelic_synthetics_script_monitor" "metrics_api_synthetic_legacy" {
  account_id       = data.newrelic_account.main-regional_1.account_id
  status           = "ENABLED"
  name             = "Metrics API Healthcheck - k8s529"
  type             = "SCRIPT_API"
  locations_public = ["US_EAST_2", "US_WEST_2"]
  period           = "EVERY_MINUTE"

  script = templatefile("${path.module}/synthetic-scripts/metrics-api.js.tftpl",
    {
      ENDPOINT = var.metrics_api_endpoint_legacy,
      LKC      = var.metrics_api_synthetic_lkc
    }
  )

  script_language      = "JAVASCRIPT"
  runtime_type         = "NODE_API"
  runtime_type_version = "16.10"

  tag {
    key    = "service"
    values = ["metrics-api"]
  }
}
```
### Actual Behavior
`terraform plan` failed with this error
```
╷
│ Error: maximum retries reached: This field is deprecated! Please use `relatedEntities` instead., Exception while fetching data (/entities[0]/relatedEntities) : Internal Error
│ 
│   with module.newrelic.newrelic_synthetics_script_monitor.metrics_api_synthetic_legacy[0],
│   on ../module/metrics-api-policy.tf line 338, in resource "newrelic_synthetics_script_monitor" "metrics_api_synthetic_legacy":
│  338: resource "newrelic_synthetics_script_monitor" "metrics_api_synthetic_legacy" {
│ 
```

**Comment by els-grazziotinf** [comment]
Had a similar error, try with `-parallelism=1` on your execution.

**Comment by sirianni** [comment]
> Had a similar error, try with `-parallelism=1` on your execution.

Thanks for the suggestion.  Why do you think it would be an intermittent issue given the error seems to be a deprecation error?

**Comment by els-grazziotinf** [comment]
I think the deprecation warning is a red herring.

It's been deprecated for a while - https://github.com/newrelic/newrelic-client-go/blob/main/internal/http/graphql.go#L93-L108

I imagine that NewRelic API server is suffering with something else and lower parallelism seems to get it going long enough. At least it did when I attempted it.

**Comment by sanderblue** [comment]
Thanks for reporting. We'll take a closer look to see what's going on here.

**Comment by pranav-new-relic** [comment]
Hi @sirianni, thank you for reporting this. We've been trying to triage this issue in the last two days, but haven't been able to make a lot of progress, given we aren't able to reproduce the issue. When we run `terraform plan` with the configuration you've specified, we don't seem to be blocked by this error.

To help triage this issue further, we would need your assistance with some more details - 
1. Do you see this happening when you perform a `terraform plan` on an entirely new configuration (or) when updates are made to some existing configuration you have? I believe it is the latter, but would like you to confirm.
2. With regard to the configuration you've attached in the description, is it a snippet of the whole configuration that is leading to this error, or is it with this exact configuration (with that one synthetics script monitor you've attached above) that you're able to see an error upon `terraform plan`?
3. If possible, would you be able to attach debug logs rendered upon `terraform plan` so we can take a deeper look into the origin of this error? This may be done by setting the environment variable `TF_LOG` to an appropriate value such as
`export TF_LOG=TRACE` (or) `export TF_LOG=DEBUG` and then running `terraform plan`. If you can do so, please make sure sensitive credentials are redacted in the debug logs - so we don't run into issues with security systems on GitHub :) it would also be helpful if you can let us know the exact version of Terraform and the version of the New Relic Terraform Provider you see this issue with.

Thank you for reading through this, and your help, in advance 🙂 

(P.S. @els-grazziotinf since you mentioned you faced this issue too, would you mind attaching the requisites I've mentioned in (3)? Multiple sources of information would help us triage this issue better. Thanks!)

**Comment by els-grazziotinf** [comment]
I'm unable to reproduce the issue, it just works again...

**Comment by pranav-new-relic** [comment]
We haven’t heard back from you in a long time so we will close the issue. If you feel this is still a valid request or bug, feel free to create a new issue.
